src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -123,6 +124,15 @@ internal bool ShouldMonitorHostHealth
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
+            // Shutdown if RunFromZipFailed
+            string runFromPackageFailedPath = Path.Combine(_applicationHostOptions.CurrentValue.ScriptPath, ScriptConstants.RunFromPackageFailedFileName);
```

Comment:
Move all this into a helper method "CheckFileSystem". Also, we should consider making this check conditional so it's only performed if environment.IsZipDeployment is true? That way we're not performing unnecessary file checks on the startup path.

Reply:
FAILED TO INITIALIZE RUN FROM PACKAGE.txt” is created with the following 2 cases:
1)	If RunFromZip = 1  and zip is corrupt/not accessible.
2)	RunFromZip = URL and the URL is not valid.

I checked the logic for environment.IsZipDeployment and it looks like it returns false if URL is invalid. So relying on that would mean, there's at least one case where result is false and the txt file is still present and we continue with starting the host.. We probably need something additional which checks for existence of the setting but not necessarily do additional validations like IsZipDeployment does..?

Reply:
Just add a flag to do uri validation, true by default, e.g. IsZipDeployment(this IEnvironment environment, bool validateUri = true). Put the conditional inside the CheckFileSystemAsync method

Reply:
It also checks validates whether the value == 1